tests/xdp/xdp_tests.cpp

```diff
@@ -0,0 +1,170 @@
+
+#define CATCH_CONFIG_RUNNER
+
+#include "catch_wrapper.hpp"
+#include "xdp_tests.h"
+#include "xdp_tests_common.h"
+
+#include <mstcpip.h>
+
+std::string _remote_ip;
+const uint16_t _reflection_port = REFLECTION_TEST_PORT;
+
+TEST_CASE("xdp_reflect_test", "[xdp_tests]")
+{
+    int error = 0;
+    uint32_t ipv6_opt = 0;
+
+    // Create a receiver socket.
+    SOCKET receiver_socket = INVALID_SOCKET;
+    receiver_socket = WSASocket(AF_INET6, SOCK_DGRAM, IPPROTO_UDP, nullptr, 0, WSA_FLAG_OVERLAPPED);
+    if (receiver_socket == INVALID_SOCKET)
+        FAIL("Failed to create receiver socket with error: " << WSAGetLastError());
+    error =
+        setsockopt(receiver_socket, IPPROTO_IPV6, IPV6_V6ONLY, reinterpret_cast<const char*>(&ipv6_opt), sizeof(ULONG));
+    if (error != 0)
+        FAIL("Could not enable dual family endpoint: " << WSAGetLastError());
+
+    // Bind it to the reflection port.
+    SOCKADDR_STORAGE recv_addr;
+    recv_addr.ss_family = AF_INET6;
+    INETADDR_SETANY((PSOCKADDR)&recv_addr);
+    ((PSOCKADDR_IN6)&recv_addr)->sin6_port = htons(_reflection_port);
+
+    error = bind(receiver_socket, (PSOCKADDR)&recv_addr, sizeof(recv_addr));
+    if (error != 0)
+        FAIL("Failed to bind receiver socket with error: " << WSAGetLastError());
+
+    WSAOVERLAPPED overlapped = {};
+
+    // Create an event handle and set up the overlapped structure.
+    overlapped.hEvent = WSACreateEvent();
+    if (overlapped.hEvent == NULL)
+        FAIL("WSACreateEvent failed with error: " << WSAGetLastError());
+
+    // Post an asynchronous receive on the receiver socket.
+    std::vector<char> recv_buffer(1024);
+    WSABUF wsa_recv_buffer{static_cast<ULONG>(recv_buffer.size()), reinterpret_cast<char*>(recv_buffer.data())};
+    uint32_t recv_flags = 0;
+    SOCKADDR_STORAGE sender_addr;
+    int sender_addr_size = sizeof(sender_addr);
+    error = WSARecvFrom(
+        receiver_socket,
+        &wsa_recv_buffer,
+        1,
+        nullptr,
+        reinterpret_cast<LPDWORD>(&recv_flags),
+        (PSOCKADDR)&sender_addr,
+        &sender_addr_size,
+        &overlapped,
+        nullptr);
+    if (error != 0) {
+        int wsaerr = WSAGetLastError();
+        if (wsaerr != WSA_IO_PENDING)
+            FAIL("WSARecvFrom failed with " << wsaerr);
+    }
+
+    // Create the sender socket for sending data.
+    SOCKET sender_socket = INVALID_SOCKET;
+    sender_socket = WSASocket(AF_INET6, SOCK_DGRAM, IPPROTO_UDP, nullptr, 0, 0);
+    if (sender_socket == INVALID_SOCKET)
+        FAIL("Failed to create sender socket with error: " << WSAGetLastError());
+    error =
+        setsockopt(sender_socket, IPPROTO_IPV6, IPV6_V6ONLY, reinterpret_cast<const char*>(&ipv6_opt), sizeof(ULONG));
+    if (error != 0)
+        FAIL("Could not enable dual family endpoint: " << WSAGetLastError());
+
+    // Initialize the remote address.
+    struct sockaddr_storage remote_address = {};
+    struct in_addr ipv4_addr;
```

Comment:
Lines 84-95 are ok, but I think it would be simpler and more maintainable to use getaddrinfo instead of having multiple if tests and using inet_pton.